clinica/pipelines/statistics_volume/statistics_volume_utils.py

```diff
@@ -0,0 +1,497 @@
+
+    Args:
+        m_file: (str) path to Matlab m file
+
+    Returns:
+        output_mat_file: (str) path to the SPM.mat file needed in SPM analysis
+    """
+    from os.path import isfile, dirname, basename, abspath, join
+    from os import system
+    import clinica.pipelines.statistics_volume.statistics_volume_utils as utls
+    from nipype.interfaces.matlab import MatlabCommand, get_matlab_command
+    import platform
+    from clinica.utils.stream import cprint
+
+    assert isinstance(m_file, str), '[Error] Argument must be a string'
+    if not isfile(m_file):
+        raise FileNotFoundError('[Error] File ' + m_file + 'does not exist')
+    assert m_file[-2:] == '.m', '[Error] ' + m_file + ' is not a Matlab file (extension must be .m)'
+
+    # Generate command line to run
+    if utls.use_spm_standalone():
+        cprint('USING SPM STANDALONE')
+        utls.delete_last_line(m_file)
+        # SPM standalone must be run directly from its root folder
+        if platform.system().lower().startswith('darwin'):
+            # Mac OS
+            cmdline = 'cd $SPMSTANDALONE_HOME && ./run_spm12.sh $MCR_HOME batch ' + m_file
+        elif platform.system().lower().startswith('linux'):
+            # Linux OS
+            cmdline = '$SPMSTANDALONE_HOME/run_spm12.sh $MCR_HOME batch ' + m_file
+        else:
+            raise SystemError('Clinica only support Mac OS and Linux')
+        system(cmdline)
+    else:
+        cprint('USING CLASSICAL SPM')
+        MatlabCommand.set_default_matlab_cmd(get_matlab_command())
+        matlab = MatlabCommand()
+        if platform.system().lower().startswith('linux'):
+            matlab.inputs.args = '-nosoftwareopengl'
+        matlab.inputs.paths = dirname(m_file)
+        matlab.inputs.script = basename(m_file)[:-2]
+        matlab.inputs.single_comp_thread = False
+        matlab.inputs.logfile = abspath('./matlab_output.log')
+        matlab.run()
+    output_mat_file = abspath(join(dirname(m_file), '..', '2_sample_t_test', 'SPM.mat'))
+    if not isfile(output_mat_file):
+        raise RuntimeError('Output matrix ' + output_mat_file + ' was not produced')
+    return output_mat_file
+
+
+def use_spm_standalone():
+    """
+        Tells if SPM standalone can be used
+    Returns:
+        True if SPM standalone is detected, False otherwise. Note that it does not guarentee that SPM (classical) is
+        up and running in the system.
+    """
+    import os
+    from os.path import isdir, expandvars
+
+    use_spm_stand = False
+    if all(elem in os.environ.keys() for elem in ['SPMSTANDALONE_HOME', 'MCR_HOME']):
+        if isdir(expandvars('$SPMSTANDALONE_HOME')) and isdir(expandvars('$MCR_HOME')):
+            use_spm_stand = True
+        else:
+            raise FileNotFoundError('[Error] $SPMSTANDALONE_HOME and $MCR_HOME are defined, but linked to non existent folder')
+    return use_spm_stand
+
+
```

Comment:
Why this function is not in `utils/spm.py`?

Reply:
Moved